iib/workers/tasks/build_create_empty_index.py

```diff
@@ -86,16 +87,30 @@ def handle_create_empty_index_request(
 
     _update_index_image_build_state(request_id, prebuild_info)
     with tempfile.TemporaryDirectory(prefix='iib-') as temp_dir:
-        if is_image_fbc(from_index_resolved):
-            err_msg = 'File-Based catalog image type is not supported yet.'
-            log.error(err_msg)
-            raise IIBError(err_msg)
         set_request_state(request_id, 'in_progress', 'Checking operators present in index image')
 
         operators = _get_present_operators(from_index_resolved, temp_dir)
-        set_request_state(request_id, 'in_progress', 'Removing operators from index image')
 
-        _opm_index_rm(temp_dir, operators, prebuild_info['binary_image'], from_index_resolved)
+        # if fbc parameter is true, create an empty FBC index image
+        # else create sqllite index image
+        if fbc:
+            log.debug('Creating empty FBC index image from %s', from_index)
+            opm_create_empty_fbc(
+                request_id=request_id,
+                temp_dir=temp_dir,
+                from_index_resolved=from_index_resolved,
+                from_index=from_index,
+                binary_image=binary_image,
+                operators=operators,
+            )
+        else:
```

Comment:
what if the from_index is FBC? Should we check it?

Reply:
opm_create_empty_fbc already checks if the from_index is fbc or not.
also there is one more case, when fbc parameter is false, but from_index given is an fbc image, that case is checked in line 107

Reply:
I don't understand why do we have the boolean flag `fbc` then, can you please elaborate?

If `opm_create_empty_fbc` can know whether the image is File-Based Catalog or not with just the information of `from_index` why do we need an additional flag here?

Isn't possible to use a similar logic to determine whether the index image is fbc or not using just `from_index` here as well? Then we wouldn't need even the check from line 107 IMO, but it's just a question :slightly_smiling_face:

Reply:
We want to specifically say that we want to create FBC image - by setting the `fbc` flag. 
Otherwise you are not able to create empty FBC index image. Because at this point you do not have any FBC image created. 

So the paths for create empty index are:
1. Input SQLite -> output SQLite
2. Input SQLite -> output FBC (flag must be set)
3. Input FBC -> output FBC  (flag must be set)

There is no backward compatibility and we can not generate SQLite from FBC image. Therefore IIB has to fail when FBC is on input and fbc flag is not set.